src/scipp/compat/xarray_compat.py

```diff
@@ -42,11 +47,50 @@ def from_xarray(obj: Union[xr.DataArray, xr.Dataset]) -> VariableLike:
         raise ValueError(f"from_xarray: cannot convert type '{type(obj)}'")
 
 
-def _var_from_xarray(xr_obj: Union[xr.Coordinate, xr.DataArray]):
+def to_xarray(obj: VariableLike) -> Union[xr.DataArray, xr.Dataset]:
+    """Convenience method to convert a scipp object into the corresponding
+    xarray object.
+
+    Parameters
+    ----------
+    obj:
+        The scipp object to convert; must be either a DataArray
+        or Dataset object.
```

Comment:
```suggestion
        The scipp object to convert.
```
The type hints should take care of the rest.